mqtt/mqtt-bridge/src/persist/memory/mod.rs

```diff
@@ -0,0 +1,215 @@
+use std::io::Error;
+use std::sync::Arc;
+
+use anyhow::Result;
+use async_trait::async_trait;
+use mqtt3::proto::Publication;
+use parking_lot::Mutex;
+use tracing::debug;
+
+use crate::persist::{
+    memory::loader::InMemoryMessageLoader, memory::waking_map::WakingMap, Key, Persist,
+};
+
+mod loader;
+mod waking_map;
+
+/// In memory persistence implementation used for the bridge
+struct InMemoryPersist {
+    state: Arc<Mutex<WakingMap>>,
+    offset: u32,
+    loader: Arc<Mutex<InMemoryMessageLoader>>,
+}
+
+#[async_trait]
+impl<'a> Persist<'a> for InMemoryPersist {
+    type Loader = InMemoryMessageLoader;
+    type Error = Error;
```

Comment:
This error type isn't used, but I had to define some error type to satisfy the trait. Is this fine?